lib/install/validate/config.go

```diff
@@ -88,18 +85,20 @@ func (e *FirewallMisconfiguredAllowedIPError) Error() string {
 
 // CheckFirewall verifies that host firewall configuration allows tether traffic and outputs results
 func (v *Validator) CheckFirewall(ctx context.Context, conf *config.VirtualContainerHostConfigSpec) {
-	defer trace.End(trace.Begin(""))
+	op := trace.FromContext(ctx, "CheckFirewall")
```

Comment:
I presume this was to avoid the cascade changes required to vicadmin if changing the signature?

Reply:
As a general pattern, I tried to have exported library functions accept a context. (I'd file this under "be liberal in what you accept'.)

If the supplied context is an operation, `trace.FromContext` will do the right thing.